data-reconciliation-app/test/e2e-test/src/index.ts

```diff
@@ -0,0 +1,132 @@
+import Api from './api';
+import https from 'https';
+import fs from 'fs';
+import { member0DataPart1, member0DataPart2, member1Data, member2Data } from './data';
+import Host from './host';
+
+const serverUrl = 'https://127.0.0.1:8000';
+const certificateStorePath = './workspace/sandbox_common';
+
+export interface DemoProps {
+    ingestUrl: string;
+    reportUrl: string;
+    proposalUrl: string;
+}
+
+export interface DemoMemberProps {
+    id: string;
+    name: string;
+    data: unknown;
+    httpsAgent: https.Agent;
+}
+
+class Demo {
+    //
+    private static readonly demoProps: DemoProps = {
+        ingestUrl: `${serverUrl}/app/ingest`,
+        reportUrl: `${serverUrl}/app/report`,
+        proposalUrl: `${serverUrl}/gov/proposals`,
+    };
+
+    private static memberDataMap = new Map([['0', member0DataPart1], ['1', member1Data], ['2', member2Data]]);
+    private static memberIds = ['0', '1', '2']
+
+    private static readonly members = Array<DemoMemberProps>();
+
+    public static async start() {
+
+        /**
+         * Change working directory to the root of the project
+         * All paths and process execution will be relative to root
+         */
+        process.chdir('../../');
+
+        this.printTestSectionHeader('🏁 Starting e2e Tests...');
+
+        this.printTestSectionHeader('🤖 Starting Host...');
+        
+        await Host.start();
+
+        for(const memberId of this.memberIds) {
+            const member = this.createMember(memberId);
+            this.members.push(member);
+        }
+
+        this.printTestSectionHeader('🔬 [TEST]: Ingestion & Reporting...');
+
+        for (const member of this.members) {
+            await Api.ingest(this.demoProps, member);
+        }
+
+        for (const member of this.members) {
+            await Api.report(this.demoProps, member);
+        }
+
+        this.printTestSectionHeader('🔬 [TEST]: Report Changes...');
+
+        const member0 = this.members[0];
+        member0.data = member0DataPart2;
+        await Api.ingest(this.demoProps, member0);
+
+        const reportItems = await Api.report(this.demoProps, member0);
+
+        if (reportItems.length !== 12) {
+            throw new Error(`🛑 [TEST FAILURE]: Unexpected number of items in the report: ${reportItems.length}`);
+        }
+        else {
+            console.log(`✅ [PASS] - ${reportItems.length} items in the report`);
+        }
+
+        const reportItem = reportItems[3];
+
+        if(reportItem.group_status !== 'IN_CONSENSUS') {
+            throw new Error(`🛑 [TEST FAILURE]: Unexpected group status ${reportItem.group_status}. Expected: IN_CONSENSUS`);
```

Comment:
To avoid making this canary app complicated, I didn't use a testing framework and instead just simply throwing an error if my assertion fails. I think this will be reflected to CI/CD actions too.